### PR TITLE
Replace sanitycheck filters with keywords

### PR DESCRIPTION
**Pull request by nashif**
Use some of the new keywords supported in testcase.yaml to replace filters.

### CODE REVIEW

File: tests/drivers/pci_enum/testcase.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 tests:
 -   test:
-        filter: CONFIG_PCI
```
Comment (galak) [inline tests/drivers/pci_enum/testcase.yaml]: shouldn't we keep the filter and remove the platform_whitelist?

**Review by galak — CHANGES_REQUESTED** [review]
looks good, curious about the PCI test and using just the filter.

File: tests/drivers/pci_enum/testcase.yaml (hunk 1 above)
Comment (nashif) [inline tests/drivers/pci_enum/testcase.yaml]: Right, but ultimate goal is to stop using filter for this type of thing. It is slow, it is also not consistent, here Qemu does not have CONFIG_PCI set by default and there is a hack in the Makefile to force it and I just noticed the test is not even running by default in Qemu. I already added support for HW feature matching, so I will be using it here for the first time, one more patch coming on top of this....

Comment (galak) [inline tests/drivers/pci_enum/testcase.yaml]: sounds good.

**Review by galak — APPROVED** [review]

**Pull request merged**